ccinput/tests/test_nwchem.py

```diff
@@ -0,0 +1,1430 @@
+
+        self.assertTrue(self.is_equivalent(REF, inp.input_file))
+        
+    def test_freq_HF(self):
+        params = {
+            "nproc": 8,
+            "mem": "10000MB",
+            "type": "Frequency Calculation",
+            "file": "Cl.xyz",
+            "software": "nwchem",
+            "method": "HF",
+            "basis_set": "3-21G",
+            "charge": "-1",
+        }
+
+        inp = self.generate_calculation(**params)
+
+        REF = """
+        TITLE "File created by ccinput"
+        start Cl
+        memory total 10000 mb
+        charge -1
+
+        geometry units angstroms
+        Cl   0.00000000   0.00000000   0.00000000
+        end
+
+        basis
+        * library 3-21G
+        end
+
+        scf
+        singlet
+        end
+
+        task scf freq
+
+
+        """
+
+        self.assertTrue(self.is_equivalent(REF, inp.input_file))
+
+    def test_freq_DFT(self):
+        params = {
+            "nproc": 8,
+            "mem": "10000MB",
+            "type": "Frequency Calculation",
+            "file": "Cl.xyz",
+            "software": "nwchem",
+            "charge": "-1",
+            "method": "B3LYP",
+            "basis_set": "6-31+G(d,p)",
+        }
+
+        inp = self.generate_calculation(**params)
+
+        REF = """
+        TITLE "File created by ccinput"
+        start Cl
+        memory total 10000 mb
+        charge -1
+
+        geometry units angstroms
+        Cl   0.00000000   0.00000000   0.00000000
+        end
+
+        basis
+        * library 6-31+G*
+        end
+
+        dft
+        xc b3lyp
+        mult 1
+        end
+
+        task dft freq
+
+
+        """
+
+        self.assertTrue(self.is_equivalent(REF, inp.input_file))
+
+    def test_freeze_bond_DFT(self):
+        params = {
+            "nproc": 8,
+            "mem": "10000MB",
+            "type": "Constrained Optimisation",
+            "file": "ethanol.xyz",
+            "software": "nwchem",
+            "charge": "0",
+            "method": "B3LYP",
+            "basis_set": "6-31+G(d,p)",
+            "constraints": "Freeze/1_2;",
+        }
+
+        inp = self.generate_calculation(**params)
+
+        REF = """
+        TITLE "File created by ccinput"
+        start ethanol
+        memory total 10000 mb
+        charge 0
+
+        geometry units angstroms
+        C   -1.31970000  -0.64380000   0.00000000
+        H   -0.96310000  -1.65260000   0.00000000
+        H   -0.96310000  -0.13940000  -0.87370000
+        H   -2.38970000  -0.64380000   0.00000000
+        C   -0.80640000   0.08220000   1.25740000
+        H   -1.16150000   1.09160000   1.25640000
+        H   -1.16470000  -0.42110000   2.13110000
+        O    0.62360000   0.07990000   1.25870000
+        H    0.94410000   0.53240000   2.04240000
+        end
+
+        basis
+        * library 6-31+G*
+        end
+
+        dft
+        xc b3lyp
+        mult 1
+        end
+
+        constraints
+        spring bond 1 2
```

Comment:
This doesn't seem to be a valid NWChem constraint. It expects two additional parameters for the spring constant and equilibrium distance. The spring constant could be a "high" value by default (whatever that is), and the equilibrium distance is already calculated from the input structure in the `Constraint` class.